tools/cfn2ts/lib/codegen.ts

```diff
@@ -305,13 +324,23 @@ export default class CodeGenerator {
    *
    * Since resolve() deep-resolves, we only need to do this once.
    */
-  private emitCloudFormationPropertiesOverride(propsType: genspec.CodeName) {
+  private emitCloudFormationPropertiesOverride(propsType: genspec.CodeName, taggable: boolean) {
     this.code.openBlock(`public get propertyOverrides(): ${propsType.className}`);
     this.code.line(`return this.untypedPropertyOverrides;`);
     this.code.closeBlock();
+    this.code.line();
 
     this.code.openBlock('protected renderProperties(properties: any): { [key: string]: any } ');
-    this.code.line(`return ${genspec.cfnMapperName(propsType).fqn}(${CORE}.resolve(properties));`);
+    this.code.line(`const resolvedProps = ${CORE}.resolve(properties);`);
+    if (taggable) {
+      this.code.line(`const rawTags: ${CORE}.Tag[] = resolvedProps.tags || [];`);
```

Comment:
Can we somehow put this code in `cdk.Resource` instead of generate it?
Maybe we can create a TaggableResource construct and basically just determine if the resource extends Resource or extends TaggableResource, instead of sprinkling generated code around